packages/rollup-core/src/types/types.ts

```diff
@@ -21,11 +18,54 @@ export interface L2ToL1Message {
 }
 
 export interface RollupTransaction {
-  nonce: number
-  gasLimit: number
-  sender: Address
+  batchIndex: number
   target: Address
   calldata: string
+  sender?: Address
+  l1MessageSender?: Address
+  gasLimit?: number
+  l1Timestamp: number
+  l1BlockNumber: number
```

Comment:
Awesome. Big fan of the `L1` prefix here. Could also be `latestL1Timestamp` but I am also definitely happy with `l1Timestamp`